lib/ansible/vars/manager.py

```diff
@@ -202,7 +203,7 @@ def _combine_and_track(data, new_data, source):
             # get role defaults (lowest precedence)
             for role in play.roles:
                 if role.public:
-                    all_vars = _combine_and_track(all_vars, role.get_default_vars(), "role '%s' defaults" % role.name)
+                    all_vars = _combine_and_track(all_vars, role.get_default_vars(), "role '%s' defaults" % role.get_name())
```

Comment:
Since `__repr__()` already calls this method, why not rely on the `repr()` interface?
```suggestion
                    all_vars = _combine_and_track(all_vars, role.get_default_vars(), f"role '{role!r}' defaults")
```